src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -515,154 +266,119 @@ public static void writeRandomSymbolValues(int size, String format, String path)
-
-        StringBuilder rs = new StringBuilder();
-        for (int digit = 0; digit < randDigits; digit++) {
-            rs.append(random.nextInt(9) + 1);
+    public static String constructString(IonStruct constraintStruct) throws Exception {
+        String constructedString;
+        String regexPattern = IonSchemaUtilities.parseTextConstraints(constraintStruct, IonSchemaUtilities.KEYWORD_REGEX);
+        Integer codePointsLengthBound = IonSchemaUtilities.parseConstraints(constraintStruct, IonSchemaUtilities.KEYWORD_CODE_POINT_LENGTH);
+        if (regexPattern != null) {
+            RgxGen rgxGen = new RgxGen (regexPattern);
+            constructedString = rgxGen.generate();
+        } else {
+            Random random = new Random();
+            // The condition of no codepoint_length constraint in ion schema.
+            if (codePointsLengthBound == null) {
+                // Preset the bound as average number 20;
+                codePointsLengthBound = random.nextInt(20);
+            }
+            StringBuilder sb = new StringBuilder();
+            for (int j = 0; j < codePointsLengthBound; j++) {
+                int codePoint;
+                int type;
+                do {
+                    codePoint = random.nextInt(DEFAULT_RANGE.get(1) - DEFAULT_RANGE.get(0) + 1) + DEFAULT_RANGE.get(0);
+                    type = Character.getType(codePoint);
+                } while (type == Character.PRIVATE_USE || type == Character.SURROGATE || type == Character.UNASSIGNED);
+                sb.appendCodePoint(codePoint);
+            }
+            constructedString = sb.toString();
         }
-        BigDecimal coefficient = new BigDecimal(rs.toString());
-        return coefficient.scaleByPowerOfTen(exp);
+        return constructedString;
     }
 
     /**
      * Construct the decimal which is conformed with the constraints provided in ISL.
-     * @param precision represents the minimum/maximum range indicating the number of digits in the unscaled value of a decimal. The minimum precision must be greater than or equal to 1.
-     * @param scale represents the minimum/maximum range indicating the number of digits to the right of the decimal point. The minimum scale must be greater than or equal to 0.
      * @return the constructed decimal.
      */
-    public static BigDecimal constructDecimalFromSchema(int precision, int scale) {
+    public static BigDecimal constructDecimal(IonStruct constraintStruct) throws Exception {
         Random random = new Random();
+        // precision represents the minimum/maximum range indicating the number of digits in the unscaled value of a decimal. The minimum precision must be greater than or equal to 1.
+        Integer precision = IonSchemaUtilities.parseConstraints(constraintStruct, IonSchemaUtilities.KEYWORD_PRECISION);
+        if (precision == null) {
+            precision = random.nextInt(Integer.MAX_VALUE);
```

Comment:
We'll probably want the default maximum to be smaller than Integer.MAX_VALUE; won't that end up generating 1GB of digits on average? Same for scale.

Reply:
> We'll probably want the default maximum to be smaller than Integer.MAX_VALUE; won't that end up generating 1GB of digits on average? Same for scale.

The default value of precision will be reset as 20, the default scale value will be a random value between -20 to 20.